src/mac_notifications/manager.py

```diff
@@ -48,14 +49,15 @@ class NotificationManager(metaclass=Singleton):
     """
 
     def __init__(self):
-        self._callback_queue: SimpleQueue = SimpleQueue()
         self._callback_executor_event: Event = Event()
         self._callback_executor_thread: CallbackExecutorThread | None = None
         self._callback_listener_process: NotificationProcess | None = None
         # Specify that once we stop our application, self.cleanup should run
         atexit.register(self.cleanup)
+
         # Specify that when we get a keyboard interrupt, this function should handle it
-        signal.signal(signal.SIGINT, handler=self.catch_keyboard_interrupt)
+        if not MACOS_NOTIFICATIONS_AS_DAEMON:
```

Comment:
this is only semi-related to the rest of my issue but it is less than ideal for an imported package to unexpectedly start trapping all `SIGINT`... even less ideal that it does not start trapping until a notification is sent.

including `macos-notifications` put my entire server in a state where it could not be killed with ctrl-C (which is not great) but only after sending the first notification (which was very unexpected).